### PR TITLE
Merge PCS pipeline into the main one

### PR DESCRIPTION
**Pull request by premun**
- Merges the PCS pipeline into the main one
- Prepares test projects (which didn't build under arcade) for easy removal of Maestro test projects

https://github.com/dotnet/arcade-services/issues/3455

### CODE REVIEW

File: eng/templates/stages/deploy.yaml [hunk 1]
```diff
@@ -21,14 +21,45 @@ stages:
...
+  - Build
+  - Approval
+
   variables:
+  - name: currentDate
+    value: $[format('{0:yyyy}-{0:MM}-{0:dd}', pipeline.startTime)]
+
   - ${{ if parameters.isProd }}:
+    - name: PublishProfile
+      value: Int
```
Comment (premun) [inline eng/templates/stages/deploy.yaml:36]: This is wrong

File: azure-pipelines.yml [hunk 2]
```diff
@@ -163,6 +177,31 @@ extends:
...
+        pool:
+          name: NetCore1ESPool-Internal
+          demands: ImageOverride -equals 1es-ubuntu-2004
+
+        steps:
+        - checkout: self
+
+        - template: /eng/templates/steps/docker-build.yml@self
+          parameters:
+            devBranchSuffix: $(devBranchSuffix)
```
Comment (dkurepa) [inline azure-pipelines.yml:191]: where are we setting this parameter? In the variable grouop?

Comment (dkurepa) [reply azure-pipelines.yml:191]: also $(containerName) bellow

**Review by dkurepa — APPROVED** [review]

**Comment by premun** [comment]
Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2604841&view=results

**Review by pavel-purma — APPROVED** [review]

**Pull request merged**